File: app/src/main/java/com/magnitudestudios/GameFace/Activities/LoginActivity.java
package com.magnitudestudios.GameFace.Activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.content.Intent;
import android.os.Bundle;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.magnitudestudios.GameFace.Fragments.LoginScreenFragment;
import com.magnitudestudios.GameFace.Fragments.SignUpScreenFragment;
import com.magnitudestudios.GameFace.Fragments.StartScreenFragment;
import com.magnitudestudios.GameFace.Interfaces.UserLoginListener;
import com.magnitudestudios.GameFace.R;

public class LoginActivity extends AppCompatActivity implements UserLoginListener {
    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_login);
        switchFragment(new StartScreenFragment());
        mAuth = FirebaseAuth.getInstance();
    }

    @Override
    protected void onStart() {
        super.onStart();
        FirebaseUser currentUser = mAuth.getCurrentUser();
        if (currentUser != null) {
            goToMainActivity();
        }
    }

    private void switchFragment(Fragment f) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.login_frame_replace, f);
        fragmentTransaction.addToBackStack("");
        fragmentTransaction.commit();
    }

    @Override
    public void onClickSignUpButton() {
        switchFragment(new SignUpScreenFragment());
    }

    @Override
    public void onClickLoginButton() {
        switchFragment(new LoginScreenFragment());
    }

    private void goToMainActivity() {
        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
        startActivity(intent);
        finish();
    }

    @Override
    public void signedInUser() {
        goToMainActivity();
    }

    @Override
    public void onBackPressed() {

    }
}

File: app/src/main/java/com/magnitudestudios/GameFace/Activities/LoginActivity.kt
package com.magnitudestudios.GameFace.Activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.firebase.auth.FirebaseAuth
import com.magnitudestudios.GameFace.Fragments.Login.LoginScreenFragment
import com.magnitudestudios.GameFace.Fragments.Login.SignUpScreenFragment
import com.magnitudestudios.GameFace.Fragments.Login.StartScreenFragment
import com.magnitudestudios.GameFace.GameFace
import com.magnitudestudios.GameFace.Interfaces.UserLoginListener
import com.magnitudestudios.GameFace.Network.FirebaseHelper
import com.magnitudestudios.GameFace.R
import com.magnitudestudios.GameFace.databinding.ActivityLoginBinding

class LoginActivity : AppCompatActivity(), UserLoginListener {
    private var mAuth: FirebaseAuth? = null
    private lateinit var binding: ActivityLoginBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
//        switchFragment(StartScreenFragment())
        mAuth = FirebaseAuth.getInstance()
    }

    override fun onStart() {
        super.onStart()
        val currentUser = mAuth!!.currentUser
        if (currentUser != null) {
            goToMainActivity()
        }
    }


    private fun goToMainActivity() {
        val intent = Intent(this@LoginActivity, MainActivity::class.java)
        startActivity(intent)
        finish()
    }

    override fun signedInUser() {
        FirebaseHelper.getUsername()

        goToMainActivity()
    }

    override fun onBackPressed() {}
}